app/controllers/admin/themes_controller.rb

```diff
@@ -401,8 +401,10 @@ def log_theme_component_enabled
   def handle_switch
     param = theme_params[:component]
     if param.to_s == "false" && @theme.component?
+      raise Discourse::InvalidParameters.new(:component) if @theme.id == SiteSetting.default_theme_id
```

Comment:
It looks like this line is repeated and you could move it out of the if

Reply:
We need to NOT raise the error if no change needs to happen -- so it would run if `(param.to_s == "true") == (@theme.component?)`

But as written above it doesn't properly handle unspecified, so I put it inside both ifs.